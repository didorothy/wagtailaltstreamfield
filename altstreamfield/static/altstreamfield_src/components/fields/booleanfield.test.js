import React from "react";
import { render, fireEvent, wait, waitForDomChange, waitForElementToBeRemoved, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import BooleanField from "./booleanfield";
import FormErrorContext from "../../context/formerror";

describe('BooleanField', () => {
    test('#minimal', () => {
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <BooleanField owner_id="struct-01" value={false} name="test" onChange={() => {}} />
        </FormErrorContext.Provider>);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#label', () => {
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <BooleanField owner_id="struct-01" value={false} name="test" onChange={() => {}} label="Test Char Field" />
        </FormErrorContext.Provider>);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#help_text', () => {
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <BooleanField owner_id="struct-01" value={false} name="test" onChange={() => {}} label="Test Char Field" help_text="Sample help text." />
        </FormErrorContext.Provider>);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#required', () => {
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <BooleanField owner_id="struct-01" value={true} name="test" onChange={() => {}} label="Test Char Field" help_text="Sample help text." required={true} />
        </FormErrorContext.Provider>);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#error', () => {
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <BooleanField owner_id="struct-01" value={false} name="test" onChange={() => {}} label="Test Char Field" help_text="Sample help text." required={true} />
        </FormErrorContext.Provider>);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('handle_change', () => {
        let current_value = false;
        let onChange = (name, value) => {
            current_value = value;
        }
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <BooleanField owner_id="struct-01" value={current_value} name="test" onChange={onChange} label="Test Char Field" help_text="Sample help text." required={true} />
        </FormErrorContext.Provider>);
        expect(manager.isValid()).toEqual(false);
        let input = container.container.getElementsByTagName('input')[0];
        let error = container.container.getElementsByClassName('error-message')[0];
        expect(error.textContent).toEqual('This field is required.')

        fireEvent.click(input);
        container.rerender(<FormErrorContext.Provider value={manager}>
            <BooleanField owner_id="struct-01" value={current_value} name="test" onChange={onChange} label="Test Char Field" help_text="Sample help text." required={true} />
        </FormErrorContext.Provider>);
        expect(current_value).toEqual(true);
        expect(manager.isValid()).toEqual(true);

        fireEvent.click(input);
        container.rerender(<FormErrorContext.Provider value={manager}>
            <BooleanField owner_id="struct-01" value={current_value} name="test" onChange={onChange} label="Test Char Field" help_text="Sample help text." required={true} />
        </FormErrorContext.Provider>);
        expect(current_value).toEqual(false);
        expect(manager.isValid()).toEqual(false);

        cleanup();
    });
});