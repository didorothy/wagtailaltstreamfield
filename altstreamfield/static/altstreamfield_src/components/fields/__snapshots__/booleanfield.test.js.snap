// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooleanField #error 1`] = `
<div>
  <div
    class="field boolean_field required error"
  >
    <label
      for="struct-01test"
    >
      Test Char Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="checkbox"
      >
        <input
          id="struct-01test"
          placeholder="Test Char Field"
          type="checkbox"
        />
      </div>
      <p
        class="help"
      >
        Sample help text.
      </p>
      <p
        class="error-message"
      >
        This field is required.
      </p>
    </div>
  </div>
</div>
`;

exports[`BooleanField #help_text 1`] = `
<div>
  <div
    class="field boolean_field"
  >
    <label
      for="struct-01test"
    >
      Test Char Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="checkbox"
      >
        <input
          id="struct-01test"
          placeholder="Test Char Field"
          type="checkbox"
        />
      </div>
      <p
        class="help"
      >
        Sample help text.
      </p>
      
    </div>
  </div>
</div>
`;

exports[`BooleanField #label 1`] = `
<div>
  <div
    class="field boolean_field"
  >
    <label
      for="struct-01test"
    >
      Test Char Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="checkbox"
      >
        <input
          id="struct-01test"
          placeholder="Test Char Field"
          type="checkbox"
        />
      </div>
      
    </div>
  </div>
</div>
`;

exports[`BooleanField #minimal 1`] = `
<div>
  <div
    class="field boolean_field"
  >
    <label
      for="struct-01test"
    >
      Test
    </label>
    <div
      class="field-content"
    >
      <div
        class="checkbox"
      >
        <input
          id="struct-01test"
          placeholder="Test"
          type="checkbox"
        />
      </div>
      
    </div>
  </div>
</div>
`;

exports[`BooleanField #required 1`] = `
<div>
  <div
    class="field boolean_field required"
  >
    <label
      for="struct-01test"
    >
      Test Char Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="checkbox"
      >
        <input
          checked=""
          id="struct-01test"
          placeholder="Test Char Field"
          type="checkbox"
        />
      </div>
      <p
        class="help"
      >
        Sample help text.
      </p>
      
    </div>
  </div>
</div>
`;
